plugins/notify/__init__.py

```diff
@@ -14,124 +18,335 @@
+        tray_icon = self.__exaile.gui.tray_icon
+        if state and not self.__tray_connection:
+            if tray_icon:
+                self.__tray_connection = tray_icon.connect(
+                    'query-tooltip', self.__on_query_tooltip)
+            else:
+                self.__try_connect_tray()
+        elif not state and self.__tray_connection:
+            if tray_icon:  # xlgui.main might already have destroyed the tray icon
+                self.__exaile.gui.tray_icon.disconnect(self.__tray_connection)
+            self.__tray_connection = None
 
-        '''
-        title = track.get_tag_display('title')
-        artist = cgi.escape(
+    def on_option_set(self, _event, settings, option):
+        if option == notifyprefs.TrayHover.name or \
+                option == 'gui/use_tray':
+            has_tray = settings.get_option('gui/use_tray')
+            shall_show_tray_hover = self.settings.tray_hover and has_tray
+            self.__set_tray_hover_state(shall_show_tray_hover)
+        elif option == notifyprefs.ShowCovers.name:
+            # delete current cover if user doesn't want to see more covers
+            if not self.settings.show_covers:
+                if self.settings.resize_covers:
+                    size = DEFAULT_ICON_SIZE[1]
+                else:
+                    size = Gtk.IconSize.DIALOG
+                new_icon = icons.MANAGER.pixbuf_from_icon_name('exaile', size)
+                self.notification.set_image_from_pixbuf(new_icon)
+
+    def __maybe_show_notification(self, summary=None, body='', icon_name=None, force_show=False):
+        """ If summary is none, don't change the Notification """
+        if summary is not None:
+            try:
+                self.notification.update(summary, body, icon_name)
+            except GLib.Error:
+                LOGGER.exception("Could not set new notification status.")
+                return
+        # although this does not look like "show_when_focused" matches
+        # "window.is_active()", it does.
+        if self.settings.show_when_focused or \
+                not self.__exaile.gui.main.window.is_active() or \
+                force_show:
+            try:
+                self.notification.show()
+            except GLib.Error:
+                LOGGER.exception("Could not set new notification status.")
+                self.__exaile.plugins.disable_plugin(__name__)
+
+    def __get_body_str(self, track):
+        artist_str = cgi.escape(
             track.get_tag_display('artist', artist_compilations=False)
         )
-        album = cgi.escape(track.get_tag_display('album'))
-
-        if artist and album:
-            body_format = self.body_artistalbum
-        elif artist:
-            body_format = self.body_artist
-        elif album:
-            body_format = self.body_album
+        album_str = cgi.escape(track.get_tag_display('album'))
+
+        if self.settings.can_show_markup:
```

Comment:
> I can't remember why we had two plugins for this, but I sort of remember hearing that some notification daemons (XFCE's maybe?) don't support formatting? I don't know whether that's still a valid issue today.

This is the place where I change the notification text body depending on whether markup is supported or not.